src/lib.rs

```diff
@@ -482,6 +482,66 @@ pub trait Rng {
         Ok(self.fill_bytes(dest))
     }
 
+    /// Fill `dest` entirely with random data.
+    /// 
+    /// This method provides a convenient way to fill a slice with random data.
+    /// 
+    /// On big-endian platforms this performs byte-swapping to ensure
+    /// portability of results from reproducible generators.
+    /// 
+    /// # Example
+    /// 
+    /// ```rust
+    /// use rand::{thread_rng, Rng};
+    /// 
+    /// let mut arr = [0i8; 20];
+    /// thread_rng().try_fill(&mut arr[..]);
+    /// ```
+    /// 
+    /// [`ErrorKind`]: enum.ErrorKind.html
+    fn fill<T: AsByteSliceMut + ?Sized>(&mut self, dest: &mut T) where Self: Sized {
+        let slice = dest.as_byte_slice_mut();
+        self.fill_bytes(slice);
+        for mut x in slice {
+            x.to_le();
```

Comment:
If I am reading this correctly this line does not do anything, because `to_le` for `u8` should be no-op. We'll need to add `to_le` method to `AsByteSliceMut` trait.

Reply:
You're right. Actually, that's what I had originally.

Reply:
Good catch, but shouldn't it just loop over `dest`?

Reply:
I don't see a trait supporting `to_le` though so it has to be done per implementation.

Reply:
Then you'll need to add another bound to allow you iterate mutably over `dest`. I think it's easier to add a method to `AsByteSliceMut`.

Reply:
I wonder if eventually we'll get a suitable trait; [for now it doesn't look like it](https://www.reddit.com/r/rust/comments/4r25qx/a_unifying_trait_for_the_integer_scalar_types/) (also nothing in the `num` crate I can see).